web/src/services/TestRun.service.ts

```diff
@@ -1,9 +1,22 @@
-import TestRunEvent, {TestRunStage} from 'models/TestRunEvent.model';
+import {TestRunStage, TraceEventType} from 'constants/TestRunEvents.constants';
+import {filter, findLastIndex} from 'lodash';
+import TestRunEvent from 'models/TestRunEvent.model';
 
 const TestRunService = () => ({
   getTestRunEventsByStage(events: TestRunEvent[], stage: TestRunStage) {
     return events.filter(event => event.stage === stage);
   },
+
+  getTestRunEventsWithLastPolling(events: TestRunEvent[]) {
```

Comment:
I like this, that we are already considering that we can have multiple polling events. GJ!